src/templates/kit/screens/ClaimDomainScreen.js

```diff
@@ -1,17 +1,26 @@
 import React, { useCallback, useState } from 'react'
 import { Info, GU, Link } from '@aragon/ui'
-import { useCheckDomain, DOMAIN_CHECK, DOMAIN_ERROR } from '../../check-domain'
-import { DomainField, Header, PrevNextFooter } from '.'
+import {
+  useCheckDomain,
+  DOMAIN_CHECK,
+  DOMAIN_ERROR,
+} from '../../../check-domain'
+import { DomainField, Header, Navigation } from '..'
 
 function ClaimDomain({
-  back,
-  data,
-  next,
-  screenIndex,
-  screens,
-  screenTitle = 'Claim a name',
+  dataKey = null,
+  screenProps: {
+    back,
+    data,
+    next,
+    screenIndex,
+    screens,
+    screenTitle = 'Claim a name',
+  },
 }) {
-  const [domain, setDomain] = useState(data.domain || '')
+  const screenData = (dataKey ? data[dataKey] : data) || {}
```

Comment:
No need to change but you could use `data = {}` as default above and there wouldn't be a need for the `|| {}`

Reply:
It also covers the case of having an undefined `data[dataKey]`.